pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,577 @@
+								Destination: regionPodHostname,
+							}]
+							if costOK {
+								if cost <= d.MaxNetworkCost {
+									numOK += 1
+								} else {
+									numNotOK += 1
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return numOK, numNotOK, nil
+}
+
+// getAccumulatedCost : calculate the accumulated cost based on the Pod's dependencies
+func getAccumulatedCost(scheduledList networkawareutil.ScheduledList, dependencyList []agv1alpha1.DependenciesInfo, nodeName string, region string,
+	zone string, costMap map[networkawareutil.CostKey]int64, no *NetworkOverhead) (int64, error) {
+
+	// keep track of the accumulated cost
+	var cost int64 = 0
+
+	// calculate accumulated shortest path
+	for _, podAllocated := range scheduledList { // For each pod already allocated
+		if podAllocated.Hostname != "" { // if hostname is not empty...
+			for _, d := range dependencyList { // For each pod dependency
+				if podAllocated.Selector == d.Workload.Selector { // If the pod allocated is an established dependency
+					if podAllocated.Hostname == nodeName { // If the Pod hostname is the node being scored
+						cost += SameHostname
+					} else { // If Nodes are not the same
+						// Get NodeInfo from pod Hostname
+						podHostname, err := no.handle.SnapshotSharedLister().NodeInfos().Get(podAllocated.Hostname)
+						if err != nil {
+							klog.ErrorS(nil, "getting pod hostname %q from Snapshot: %v", podHostname, err)
+							return cost, err
+						}
+						// Get zone and region from Pod Hostname
+						regionPodHostname := networkawareutil.GetNodeRegion(podHostname.Node())
+						zonePodHostname := networkawareutil.GetNodeZone(podHostname.Node())
+
+						if regionPodHostname == "" && zonePodHostname == "" { // Node has no zone and region defined
+							cost += MaxCost
+						} else if region == regionPodHostname { // If Nodes belong to the same region
+							if zone == zonePodHostname { // If Nodes belong to the same zone
+								cost += SameZone
+							} else { // belong to a different zone
+								value, ok := costMap[networkawareutil.CostKey{ // Retrieve the cost from the map (origin: zone, destination: pod zoneHostname)
+									Origin:      zone, // Time Complexity: O(1)
+									Destination: zonePodHostname,
+								}]
+								if ok {
+									cost += value // Add the cost to the sum
+								} else {
+									cost += MaxCost
+								}
+							}
+						} else { // belong to a different region
+							value, ok := costMap[networkawareutil.CostKey{ // Retrieve the cost from the map (origin: region, destination: pod regionHostname)
+								Origin:      region, // Time Complexity: O(1)
+								Destination: regionPodHostname,
+							}]
+							if ok {
+								cost += value // Add the cost to the sum
+							} else {
+								cost += MaxCost
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return cost, nil
+}
+
+func findAppGroupNetworkOverhead(agName string, no *NetworkOverhead) (*agv1alpha1.AppGroup, error) {
```

Comment:
better passing a simple lister instead the plugin?

```suggestion
func findAppGroupNetworkOverhead(agName string, agLister aglisters.AppGroupLister) (*agv1alpha1.AppGroup, error) {
```